substrate/frame/asset-conversion/src/lib.rs

```diff
@@ -978,42 +1142,53 @@ pub mod pallet {
 		}
 
 		/// Leading to an amount at the end of a `path`, get the required amounts in.
-		pub(crate) fn get_amounts_in(
-			amount_out: &T::AssetBalance,
-			path: &BoundedVec<T::MultiAssetId, T::MaxSwapPathLength>,
-		) -> Result<Vec<T::AssetBalance>, DispatchError> {
-			let mut amounts: Vec<T::AssetBalance> = vec![*amount_out];
-
-			for assets_pair in path.windows(2).rev() {
-				if let [asset1, asset2] = assets_pair {
-					let (reserve_in, reserve_out) = Self::get_reserves(asset1, asset2)?;
-					let prev_amount = amounts.last().expect("Always has at least one element");
-					let amount_in = Self::get_amount_in(prev_amount, &reserve_in, &reserve_out)?;
-					amounts.push(amount_in);
-				}
+		pub(crate) fn balance_path_from_amount_out(
```

Comment:
Similarly here I'd appreciate some comments explaining the logic

Reply:
I would also appreciate a doc comment since I think the one left is not the correct one/enough